rdr_service/genomic/genomic_job_components.py

```diff
@@ -1694,6 +1697,63 @@ def calculate_contamination_category(self, sample_id, raw_contamination, member:
 
         return contamination_category
 
+    def _set_metrics_data_file_paths(self, row: dict):
+        cvl_site_bucket_map = config.getSettingJson(config.GENOMIC_CVL_SITE_BUCKET_MAP)
+        prefix_map = config.getSettingJson(config.GENOMIC_CVL_SITE_PREFIX_MAP)
+        site_id = self.file_obj.fileName.split('_')[0].lower()
+        cvl_bucket = cvl_site_bucket_map[site_id]
+        gc_bucket = config.getSetting(cvl_bucket)
+
+        if row['genometype'] == 'aou_array':
+            for file_def in array_file_types_attributes:
+                if file_def['required']:
+                    if 'idat' in file_def["file_type"]:
+                        file_path = f'gs://{gc_bucket}/Genotyping_sample_raw_data/{row["chipwellbarcode"]}' + \
+                                    f'_{file_def["file_type"]}'
+                    else:
+                        file_path = f'gs://{gc_bucket}/Genotyping_sample_raw_data/{row["chipwellbarcode"]}.' + \
+                                    f'{file_def["file_type"]}'
+                    row[file_def['file_path_attribute']] = file_path
+                    row[file_def['file_received_attribute']] = 1
+            # Need to set wgs file received flags to 0 to prevent an error when ingesting updated AW2
+            for file_def in wgs_file_types_attributes:
+                if file_def['required']:
+                    row[file_def['file_received_attribute']] = 0
+                elif 'gvcf' in file_def['file_type']:
+                    row[file_def['file_received_attribute']] = 0
+
+        elif row['genometype'] == 'aou_wgs':
+            for file_def in wgs_file_types_attributes:
+                if file_def['required']:
+                    file_path = f'gs://{gc_bucket}/{prefix_map[site_id][file_def["file_type"]]}/{site_id.upper()}_' + \
+                                f'{row["biobankid"]}_{row["sampleid"]}_{row["limsid"]}_1.{file_def["file_type"]}'
+                    row[file_def['file_path_attribute']] = file_path
+                    row[file_def['file_received_attribute']] = 1
+                # gvcf files not required but needed for AW3 generation
+                elif 'gvcf' in file_def['file_type']:
+                    file_path = f'gs://{gc_bucket}/{prefix_map[site_id][file_def["file_type"]]}/{site_id.upper()}_' + \
+                                f'{row["biobankid"]}_{row["sampleid"]}_{row["limsid"]}_1.{file_def["file_type"]}'
+                    row[file_def['file_path_attribute']] = file_path
+                    row[file_def['file_received_attribute']] = 1
+            # Need to set array file received flags to 0 to prevent an error when ingesting updated AW2
+            for file_def in array_file_types_attributes:
+                if file_def['required']:
+                    row[file_def['file_received_attribute']] = 0
+
+        return row
+
+    def _update_member_state_after_aw2(self, member: GenomicSetMember):
```

Comment:
I appreciate the re-use of the state machine here, but in practice this part of our AW2 job takes forever. I know it wasn't part of the original request, but it would be super helpful if you could refactor this part of the job to update the workflow state on a set of member IDs in bulk? You can create a new member_dao method if needed, pass it either the GEM_READY or CVL_READY workflow state and the list of IDs and just bulk update them. Let me know if you have any questions about this.

Reply:
This is still used for the method GenomicFileIngester.ingest_single_aw2_row_for_member, but for GenomicFileIngester._process_gc_metrics_data_for_insert I added a member_dao method to perform a bulk update.